ironfish/src/node.ts

```diff
@@ -246,6 +245,17 @@ export class FullNode {
       config.setOverride('bootstrapNodes', networkDefinition.bootstrapNodes)
     }
 
+    if (config.get('generateNewIdentity')) {
```

Comment:
nit suggestion you can either take it or leave it

```javascript
    const createNewIdentity =
      config.get('generateNewIdentity') || !isHexSecretKey(internal.get('networkIdentity'))

    if (createNewIdentity) {
      const newId = new BoxKeyPair()
      internal.set('networkIdentity', newId.secretKey.toString('hex'))
      await internal.save()
    }

    privateIdentity = BoxKeyPair.fromHex(internal.get('networkIdentity'))
```

Reply:
actually don't know the implications here of not using `save()` if we already have an identity (possibly overrides wouldn't get saved 🤷🏻 ). so could be safer just to keep it the same

Reply:
It's a bit undefined whether passing in a private identity to a node should save to the internal.json or not, but I felt like it would be confusing if the identity in the node doesn't match the internal.json file (that file should basically be "per-node" anyway). So I think for now I'd like to save in all cases. But I'll see about pulling out the `internal.get` function

Reply:
Actually, I think the logic is okay as-is, so I'm going to merge. We want to use `privateIdentity` regardless of `internalIdentity` if it's set, unless generateNewIdentity is also set. Otherwise we fall back to internalIdentity (if it's valid).